libr/core/cmd_section.c

```diff
@@ -232,10 +264,10 @@ static int cmd_section(void *data, const char *input) {
 		}
 		break;
 	case '\0':
-		r_io_section_list (core->io, core->offset, 0);
+		print_sections (core->bin, core->offset, R_FALSE);
```

Comment:
as we discussed on irc. this is really a breaking change. im aware of this, but we should do that change carefully, mainly because it breaks what everyone has been using sicne now. and removing sections from the S command and putting the program headers in it will break a lot of things. maybe we should just add another command to cope with program headers, or maybe we should just keep using sections as what its now, setting the flags and such, but only allocating some of them to io, but all of them should be displayed in the S command. otherwise the logic behind S* subcommands will need to be moved somewhere else. What about adding “nop” sections, or adding a config var to change that behaviour?

> On 25 Aug 2015, at 16:50, Riccardo Schirone <notifications@github.com> wrote:
> 
> In libr/core/cmd_section.c <https://github.com/radare/radare2/pull/3160#discussion_r37874314>:
> 
> > @@ -232,10 +264,10 @@ static int cmd_section(void *data, const char *input) {
> >  		}
> >  		break;
> >  	case '\0':
> > -		r_io_section_list (core->io, core->offset, 0);
> > +		print_sections (core->bin, core->offset, R_FALSE);
> Looking again at this, i think using print_sections instead of r_io_section_list would break all the 'S' commands ....
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/radare/radare2/pull/3160/files#r37874314>.
>